### ISSUE TITLE
Uncaught Error: another server is already running, but failed to connect to it

### ISSUE THREAD

**Issue by christaus** [jedwards1211/import-it#6]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.57.0 x64
**Electron**: 9.4.4
**OS**: Unknown Windows version
**Thrown From**: [import-it](https://github.com/jedwards1211/import-it) package 1.1.0

### Stack Trace

Uncaught Error: another server is already running, but failed to connect to it

```
At events.js:200

Error: another server is already running, but failed to connect to it
    at /packages/import-it/node_modules/dude-wheres-my-module/Client.js:93:27)
    at Object.onceWrapper (events.js:313:26)
    at ChildProcess.emit (events.js:223:5)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
```

### Commands

```
     -3:49.3.0 typescript:activate (atom-workspace.workspace.scrollbars-visible-always)
     -3:38.4.0 editor:newline (input.hidden-input)
  3x -3:35.1.0 core:move-down (input.hidden-input)
     -3:33.6.0 autocomplete-plus:confirm (input.hidden-input)
  9x -3:30.2.0 core:move-left (input.hidden-input)
     -3:25.6.0 editor:move-to-end-of-screen-line (input.hidden-input)
     -3:01.9.0 import-it:add-imports (span.syntax--source.syntax--js)
     -3:00.7.0 linter:lint (atom-text-editor.editor.typescript-editor)
 26x -2:48.4.0 core:backspace (input.hidden-input)
     -2:37 editor:consolidate-selections (input.hidden-input)
     -2:37 core:cancel (input.hidden-input)
  2x -1:07.6.0 core:backspace (input.hidden-input)
     -0:18 datatip:toggle (input.hidden-input)
     -0:17.7.0 import-it:add-imports (input.hidden-input)
```

### Non-Core Packages

```
atom-beautify 0.33.4 
atom-file-icons 0.7.1 
atom-ide-base 2.6.0 
atom-ide-datatip 0.24.1 
atom-ide-definitions 0.4.1 
atom-ide-hyperclick 1.0.10 
atom-ide-javascript 1.5.0 
atom-ide-markdown-service 2.1.0 
atom-ide-outline 2.3.4 
atom-ide-signature-help 0.15.3 
atom-typescript 14.2.1 
autoclose-html 0.23.0 
autocomplete-paths 2.15.2 
autocomplete-project-paths 2.2.0 
busy-signal 2.0.1 
color-picker 2.3.0 
editorconfig 2.6.1 
gist-it 0.9.2 
git-history 3.3.0 
git-time-machine 2.1.0 
highlight-selected 0.17.0 
ide-python 1.8.0 
import-it 1.1.0 
intentions 1.1.5 
javascript-drag-import 0.1.13 
less-than-slash 0.20.0 
linter 3.3.1 
linter-eslint 8.6.6 
linter-ui-default 3.3.1 
markdown-pdf 2.3.3 
minimap 4.39.14 
minimap-highlight-selected 4.6.5 
pigments 0.40.6 
platformio-ide-terminal 2.10.1 
prettier-atom 0.60.1 
project-manager 3.3.8 
python-black 0.7.0 
split-diff 1.6.1 
tool-bar 1.4.2 
tool-bar-atom 1.7.1 
```

**Comment by jedwards1211** [comment]
Sorry for the delay 😕 
I haven't tested much on windows, let me try it...

For now you may be able to work around this if you run the Stop Server or Kill Server command in the extension menu.

**Comment by christaus** [comment]
A delay? No, you replied, just for that: thank you.

I would be glad to test it but atom and many extensions is just something I
occasionnaly use on old computers.
Most of time I do use jetbrains due to many options opensource software
don't provide.

What I do like on jetbrains suite is the auto import, when using variables
or a known lib, it provide the way to auto-import the needed libs /
variables just one click while coding whatever you are using javascript,
python, java or kotlin (the main languages I do use it for sure do the same
for more).

Anyway, atom could be something interesting, perhaps in a near futur, I
will seriously retest it soon, even if most extensions work better on linux
(I cannot use, I have a too recent computer :-P).

Thank you for your feedback and the work you provide to everyone.

Le jeu. 3 juin 2021 à 01:32, Andy Edwards ***@***.***> a
écrit :

**Comment by jedwards1211** [comment]
I see, admittedly this extension will suffer a bit since I use VSCode these days, rather than Atom.

But, this is really a problem with my underlying `dude-wheres-my-module` package, and could be a problem for my VSCode extension as well.  The `dude-wheres-my-module` client finds the file that a server is already running, but fails to connect a socket to it...might just be that the server died but the files are still there.  Probably in that case I need to make it try to kill and restart the server automatically.  Need to double check what happens in that case.

**Comment by christaus** [comment]
You told a server, does it use the socket 8000 ? If yes, this is the port on wich I were using a dev server for  rest api.

**Comment by jedwards1211** [comment]
No, it uses a unix socket

**Comment by jedwards1211** [comment]
Which, now that I look at it, using a unix socket probably won't work on windows in any case, seems like I'll have to make it use a pipe instead on windows.  Sorry I failed to think about that...I'll see if I can get an alternative working on windows and get back to you

**Comment by christaus** [comment]
Absolutly, Atom is my 2nd editor, after jetbrain suite. And the main
computer I do use is running windows. Just email me or issue me on my
github profile when ready.
Best regards, Chris

Le mer. 9 juin 2021 à 18:37, Andy Edwards ***@***.***> a
écrit :

**Comment by jedwards1211** [comment]
Got `dude-wheres-my-module` working on Windows.  There was a snag with it using an old version of `fsevents` that isn't compatible with Atom anymore; hopefully I can get that fixed tomorrow and release a new version of `import-it`.

**Comment by jedwards1211** [comment]
Okay I published version 1.2.0 of `import-it`, it may work, but I haven't tested it on Windows yet.  Will have to test it tomorrow on my Windows VM

**Comment by christaus** [comment]
Thank you for the fix. On my side I cannot test anymore because I don't have anymore computer running windows :D.
